jax3dp3/c2f.py

```diff
@@ -77,7 +77,7 @@ def score_poses_batched(
     filter_size
 ):
     all_weights = None
-    batches = jnp.array_split(pose_proposals, pose_proposals.shape[0] // 2000)
+    batches = jnp.array_split(pose_proposals, np.ceil(pose_proposals.shape[0] / 1000).astype(int))
```

Comment:
Since we are switching to a more coherent trace representation (take a look at trace.py to see what i'm talking about). Can you create a version of `score_traces` (a function in that file) that works in batch mode?